### PR TITLE
JDK16 requires JVM_IsCDSDumpingEnabled

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Enabled `JVM_IsCDSDumpingEnabled` for `JDK15+` from previously `JDK15` only.

Note: this fixes following error:
```
01:08:38  /home/jenkins/workspace/Build_JDKnext_x86-64_linux_Personal/build/linux-x86_64-server-release/support/native/java.base/libjava/CDS.o: In function `Java_jdk_internal_misc_CDS_isDumpingArchive0':
01:08:38  /home/jenkins/workspace/Build_JDKnext_x86-64_linux_Personal/./src/java.base/share/native/libjava/CDS.c:49: undefined reference to `JVM_IsCDSDumpingEnabled'
01:08:38  collect2: error: ld returned 1 exit status
```
`JDK16` with this PR builds and `java -version` output:
```
openjdk version "16-internal" 2021-03-16
OpenJDK Runtime Environment (build 16-internal+0-adhoc.fengjcaibmcom.openj9-openjdk-jdk)
Eclipse OpenJ9 VM (build jdk16jvmiscdsde-e9d73ae0f6, JRE 16 Mac OS X amd64-64-Bit Compressed References 20201020_000000 (JIT enabled, AOT enabled)
OpenJ9   - e9d73ae0f6
OMR      - b7e8a5d78
JCL      - bf162e351b0 based on jdk-16+20)
```

Issue https://github.com/eclipse/openj9/issues/10935

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

**Comment by JasonFengJ9** [comment]
@pshipton could you please review?

**Comment by pshipton** [comment]
jenkins compile zlinux jdk15

**Comment by pshipton** [comment]
jenkins compile zlinux jdknext depends ibmruntimes/openj9-openjdk-jdk#openj9-staging

**Comment by pshipton** [comment]
jenkins compile win jdk15

**Comment by keithc-ca** [comment]
I think `JVM_IsDynamicDumpingEnabled()` should be removed now, not in a separate later pull request.

**Comment by keithc-ca** [comment]
@JasonFengJ9 You should first create an issue for things like this (if it doesn't already exists) so we don't have multiple people spending time on fixes (that was my hope when I created #10935).

**Comment by JasonFengJ9** [comment]
> I think JVM_IsDynamicDumpingEnabled() should be removed now, not in a separate later pull request.

What happens if the JDK16 OpenJDK abuild doesn't promote in a timely fashion?

**Comment by keithc-ca** [comment]
> > I think JVM_IsDynamicDumpingEnabled() should be removed now, not in a separate later pull request.
> 
> What happens if the JDK16 OpenJDK abuild doesn't promote in a timely fashion?

I see no danger of that happening. There are already several examples of successful builds of jdknext#openj9-staging.

**Comment by pshipton** [comment]
Although openj9-staging is building, we haven't done any acceptance type builds to ensure all platforms are good and passing the promotion criteria. It is possible to do both steps at once: run the acceptance criteria in the PR, merge the PR and promote openj9-staging at the same time. However, given this PR is ready to go, will unblock the acceptance build, and the machines should be busy later today running the Oct release builds, I'm going to go ahead and merge this as-is. The jdknext acceptance can run on it's schedule and we can do any necessary cleanup later.

**Review by pshipton — APPROVED** [review]

**Comment by JasonFengJ9** [comment]
> I see no danger of that happening. There are already several examples of successful builds of jdknext#openj9-staging.

If such PR renaming JVM method in question is merged, JDK16 builds will fail until OpenJDK abuild promotion.
In additional, there is a chance that new changes come into `openj9-staging` branch while an `OpenJ9` PR is in progress. It might need another trivial fix or require reasonable size work to block abuild for a while. Hypothetically the PR in question can be reverted to restore `JVM_IsDynamicDumpingEnabled` but I just don't think it is a good practice to take unnecessary risk for breaking existing JDK16 builds though I agree it is low.

**Comment by pshipton** [comment]
What Keith suggests is feasible. It doesn't matter what other changes come into openj9-staging while the PR build is running. If the PR build passed, we'd push the tested SHA, not the latest changes. If the PR build failed, we wouldn't merge anything. The builds would be kept in a proven working state.

**Comment by JasonFengJ9** [comment]
> If the PR build passed, we'd push the tested SHA, not the latest changes. 

Does the PR build (running in this PR for example) push the OpenJDK `openj9-staging` content? I thought it is OpenJDK abuild. Please clarify it is wrong impression.

**Comment by keithc-ca** [comment]
The PR build (for jdk16) used the openj9-staging branch. If subsequent changes to that branch were problematic, we could push the good commit (SHA used in the successful build) to the openj9 branch, leaving the troublesome changes for later.

**Comment by JasonFengJ9** [comment]
> we could push the good commit (SHA used in the successful build) to the openj9 branch, leaving the troublesome changes for later.

Is this a manual action or by the PR builds, i.e., the PR builds record good SHA in `openj9-staging` branch and picked up later by OpenJDK abuild later?

**Comment by pshipton** [comment]
It's manual. The committer needs to run PR testing equivalent to the acceptance build, and then push the right SHA to promote the tested material in addition to merging the PR.

**Comment by JasonFengJ9** [comment]
Okay, that matches my impression.

**Pull request merged**
